remove use AddressTrait from ModuleInfo macro

LukaszRozmej opened the pull request:
# Description
Remove unnecessary use

## Linked Issues
- Fixes #635

LukaszRozmej committed:
remove use AddressTrait from ModuleInfo macro

neysofu reviewed (changes requested):
I'll create a regression test before merging

vlopes11 committed:
fix: shadow trait imports for macros

This commit introduces a pattern to shadow trait imports under macro
implementations.

This is considered a good practice as such imports will never conflict
with other imports performed by the consumer of the macro.

related issue: #635

Commit:
Regression test

Commit:
Fix regression test
